### PR TITLE
Avishag/profile page

### PR DESCRIPTION
**Pull request by avishag-gvili**

### CODE REVIEW

**Review by dvory-git — APPROVED** [review]
add page profile is OK

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+  const generateTableData = (profile) => {
+    if (!profile || !profile.listWebsites || profile.listWebsites.length === 0) {
+      return { headers: [], rows: [] };
+    }
+    const rows = profile.listWebsites
+      .filter(website => website && website.websiteId)
+      .map(website => ({
+        id: website.websiteId ? website.websiteId._id : '',
+        name: (editRowId === website.websiteId._id && editedRows) ? editedRows.name : website.websiteId.name,
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:168]: save editRowId === website.websiteId._id && editedRows into a variable and use it

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        url: (editRowId === website.websiteId._id && editedRows) ? editedRows.url : website.websiteId.url,
+        status: website.status || '',
+        limitedMinutes: (editRowId === website.websiteId._id && editedRows)
+          ? (editedRows.limitedMinutes === 0 ? '-' : editedRows.limitedMinutes)
+          : (website.limitedMinutes === 0 ? '-' : website.limitedMinutes),
+        Actions: 'Actions',
+      }));
+
+    return {
+      headers: ['name', 'url', 'status', 'limitedMinutes', 'Actions'],
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:178]: move all texts to constants file

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+  };
+  return (
+    <div className="profile-list-container">
+      <div className="profile-list-select-wrapper">
+        <div className="component-add">
+          <AddProfileComponent />
+        </div>
+        <Select
+          options={profiles.map((profile) => ({ text: profile.profileName, value: profile._id }))}
+          onChange={handleProfileSelect}
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:190]: the select still doesnt display its selected value, please treat it

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+          className="profile-list-select"
+        />
+
+
+
+        {time!==0 &&
+          <div className='timer'>
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:197]: timer should be displayed in the profiles table, next to the profile who started the timer

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+          <div className="component-update">
+            <UpdateProfileComponent profile={selectedProfile} onProfileUpdated={handleProfileUpdated} />
+          </div>}
+      </div>
+      {loading ? (
+        <div className="profile-list-loading">Loading profiles...</div>
+      ) : selectedProfile ? (
+        <div className="profile-list-details">
+          <h1>
+            Hello! You have selected your {selectedProfile.profileName} profile that
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:210]: please change the text to be:
Selected profile name
start hour:
end hour: 
sites grid

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 6]
```diff
@@ -0,0 +1,408 @@
+import React, { useState, useEffect } from 'react';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Alert from '@mui/material/Alert';
+import Grid from '@mui/material/Grid';
+import Box from '@mui/material/Box';
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:9]: join all @mui/material to one line

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 7]
```diff
@@ -0,0 +1,408 @@
...
+import GenericInput from '../../stories/GenericInput/genericInput.jsx';
+import Select from '../../stories/Select/Select.jsx';
+import GenericButton from '../../stories/Button/GenericButton.jsx';
+import { updateProfileApi, deleteProfileApi } from '../../services/profileService.js';
+import { createWebsite, updateWebsite } from '../../services/websiteService.js';
+import { deleteProfile } from '../../redux/profile/profile.slice.js';
+import { useDispatch } from 'react-redux';
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:16]: reorder your imports

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+import ToastMessage from '../../stories/Toast/ToastMessage.jsx';
+export default function UpdateProfileComponent({ profile, onProfileUpdated }) {
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:18]: add line break before the export

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 9]
```diff
@@ -0,0 +1,408 @@
...
+    const dispatch = useDispatch();
+    const [open, setOpen] = useState(false);
+    const [confirmDeleteOpen, setConfirmDeleteOpen] = useState(false);
+    const [toastOpen, setToastOpen] = useState(false);
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:22]: as I already wrote in addProfile.jsx, the toast messages management shouldnt be here but be common to he whole app. please check what is the conventional way and fix it.

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+    const [toastType, setToastType] = useState('success');
+    const [toastMessage, setToastMessage] = useState('');
+    const [formData, setFormData] = useState({
+        userId: '',
+        profileName: 'Default Profile Name',
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:27]: there is no default name to profile, the name is required so use empty string

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 11]
```diff
@@ -0,0 +1,231 @@
...
+          ? (editedRows.limitedMinutes === 0 ? '-' : editedRows.limitedMinutes)
+          : (website.limitedMinutes === 0 ? '-' : website.limitedMinutes),
+        Actions: 'Actions',
+      }));
+
+    return {
+      headers: ['name', 'url', 'status', 'limitedMinutes', 'Actions'],
+      rows: rows,
+    };
+  };
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:181]: please check which of your function should be memoized with useCallback()

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 12]
```diff
@@ -0,0 +1,408 @@
...
+        setConfirmDeleteOpen(false);
+        setToastOpen(false);
+    };
+
+    const getStatusOptions = () => {
+        const { statusBlockedSites } = formData;
+
+        if (statusBlockedSites === 'black list') {
+            return [
+                { text: 'Open', value: 'open' },
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:82]: use common enum with addProfile

File: client-side/src/components/profileComponents/addProfileComponent.jsx [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+      urlTimeLimit: 0,
+      urlStatus: '',
+    });
+    setArrUrl([]);
+    setOpen(false);
+  };
+
+  const options = {
+    black: [
+      { text: 'open', value: 'open' },
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/addProfileComponent.jsx:69]: use common enum with updateProfile

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 14]
```diff
@@ -0,0 +1,408 @@
...
+            }
+        }
+    };
+
+    const isFormValid = () => {
+        return formData.profileName.trim() !== '' &&
+            formData.timeProfile.timeStart.trim() !== '' &&
+            formData.timeProfile.timeEnd.trim() !== '';
+    };
+
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:242]: the component is too long, please move logic to profileUtil.js
as well, check which of your functions should be memoized with useCallback()

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 15]
```diff
@@ -0,0 +1,408 @@
...
+    return (
+        <div>
+            <GenericButton label='Edit Profile'onClick={handleClickOpen} size = "medium" className="profile-list-button"/>
+            <Dialog open={open} onClose={handleClose} fullWidth maxWidth="md">
+                <DialogTitle>Edit Profile</DialogTitle>
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:247]: move all texts to constants file

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 16]
```diff
@@ -0,0 +1,408 @@
...
+                                        value={formData.profileName}
+                                        onChange={(e) => handleFieldChange(e)}
+                                        width='100%'
+                                    />
+                                </Box>
+                            </Grid>
+                            <Grid item xs={1} sm={2}>
+                                <Box mt={3.2}>
+                                    <GenericInput
+                                        name="timeStart"
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:265]: add labels to all inputs (take them from constants file)

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 17]
```diff
@@ -0,0 +1,408 @@
...
+                                        onChange={(e) => handleFieldChange(e)}
+                                        options={[
+                                            { text: 'Black List', value: 'black list' },
+                                            { text: 'White List', value: 'white list' }
+                                        ]}
+                                        widthOfSelect='4.5cm'
+                                    />
+                                </Box>
+                            </Grid>
+                            <Grid item xs={12}>
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:299]: edit profile should enable only editing of profile name, start & end times, and status. editing websites should be only through inline edit in the grid

File: client-side/src/styles/profilePageStyle.scss [hunk 18]
```diff
@@ -0,0 +1,125 @@
+.profile-list-container {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: flex-start;
+    width: 100%;
+    padding: 20px;
```
Comment (SariGlick) [inline client-side/src/styles/profilePageStyle.scss:7]: replace px with %

File: client-side/src/styles/profilePageStyle.scss [hunk 19]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+.profile-list-select-wrapper {
+    display: flex;
+    justify-content: center;
+    width: 100%;
+    margin-bottom: 20px;
```
Comment (SariGlick) [inline client-side/src/styles/profilePageStyle.scss:14]: replace px with %

File: client-side/src/styles/profilePageStyle.scss [hunk 20]
```diff
@@ -0,0 +1,125 @@
...
+.profile-list-select {
+    position: absolute;
+    top:25px;
+    width: 200px;
+}
+
+
+.profile-list-loading {
+    font-size: 1.2em;
+    color: #333;
```
Comment (SariGlick) [inline client-side/src/styles/profilePageStyle.scss:26]: move colors to variables

File: client-side/src/styles/profilePageStyle.scss [hunk 21]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+
+.profile-list-details {
+    text-align: center;
+    width: 100%;
+    max-width: 1200px;
```
Comment (SariGlick) [inline client-side/src/styles/profilePageStyle.scss:33]: replace px with %, in all occurrences

File: server-side/.env [hunk 22]
```diff
@@ -1,5 +1,5 @@
-PORT=3000
+PORT=5003
 password=EuACQK5n92JI3vn4
-DB_URL="mongodb://127.0.0.1:27017/TimeOut"
+DB_URL=mongodb+srv://sh3000444:EuACQK5n92JI3vn4@cluster0.2ttcjcg.mongodb.net/?retryWrites=true&w=majority&appName=Cluster0
```
Comment (SariGlick) [inline server-side/.env:3]: remove this change

File: server-side/.env [hunk 23]
```diff
@@ -1,5 +1,5 @@
-PORT=3000
+PORT=5003
```
Comment (SariGlick) [inline server-side/.env:1]: why did you change the default port?

File: server-side/controllers/profile.controller.js [hunk 24]
```diff
@@ -1,82 +1,74 @@
...
-export const getAllProfiles = async (req, res, next) => {
+import Profile from '../models/profile.model.js';
+export const getAllProfiles = async (req, res) => {
     try {
-        const profiles = await Profiles.find().populate('limitedWebsites.websiteId blockedSites').select('-__v');
+        const profiles = await Profile.find();
         res.json(profiles);
     } catch (err) {
-        next({message:err.message,status:500})
+        res.status(500).json({ message: err.message });
```
Comment (SariGlick) [inline server-side/controllers/profile.controller.js:7]: add return to all return statements in catch blocks

**Review by SariGlick — CHANGES_REQUESTED** [review]

**Review by dvory-git — APPROVED** [review]

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 25]
```diff
@@ -0,0 +1,329 @@
...
+import { useSnackbar } from 'notistack';
+import Select from '../../stories/Select/Select.jsx';
+import TableComponent from '../../stories/table/TableComponent.jsx';
+import ToastMessage from '../../stories/Toast/ToastMessage.jsx';
+import { updateProfileApi, getProfilesByUserId } from '../../services/profileService.js';
+import { deleteWebsite, updateWebsite, createWebsite } from '../../services/websiteService.js';
+import { useAppSelector } from '../../redux/store.jsx';
+import { setProfiles, updateProfile } from '../../redux/profile/profile.slice.js';
+import { selectProfile } from '../../redux/profile/profile.selector.js';
+import { Delete as DeleteIcon, Edit as EditIcon, Save as SaveIcon, Cancel as CancelIcon } from '@mui/icons-material';
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:12]: move this import above the internal imports

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 26]
```diff
@@ -0,0 +1,206 @@
...
+            }
+        }
+    }, [dispatch, profile]);
+
+    return (
+        <div>
+            <Tooltip title={TOOLTIP_TEXTS.EDIT_PROFILE}>
+                <GenericButton label={BUTTON_LABELS.EDIT_PROFILE} onClick={handleClickOpen} size="medium" className="profile-list-button" />
+            </Tooltip>
+            <Dialog open={open} onClose={handleClose} fullWidth maxWidth="md">
```
Comment (SariGlick) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:119]: the component is too complicated, please break it into smaller components

File: client-side/src/stories/stylies.scss [hunk 27]
```diff
@@ -0,0 +1,5 @@
+$background-class: #ffffff;
```
Comment (SariGlick) [inline client-side/src/stories/stylies.scss:1]: remove this file

**Review by SariGlick — CHANGES_REQUESTED** [review]

File: client-side/src/components/profileComponents/profilePageComponent.jsx (hunk 5 above)
Comment (EA171) [reply client-side/src/components/profileComponents/profilePageComponent.jsx:307]: also move the texts to contants file

File: server-side/controllers/profile.controller.js [hunk 28]
```diff
@@ -1,77 +1,81 @@
...
-export const createProfile = async (req, res, next) => {
+export const createProfile = async (req, res) => {
+    const newProfile = new Profile(req.body);
     try {
-        const newProfile = new Profiles(req.body);
-        await newProfile.validate();
-        await newProfile.save();
-        res.status(201).json(newProfile);
+        const savedProfile = await newProfile.save();
+        res.status(201).json(savedProfile);
```
Comment (EA171) [inline server-side/controllers/profile.controller.js:16]: return is missing - add it in all occurens

File: client-side/src/axios/middleware.js [hunk 29]
```diff
@@ -9,6 +9,7 @@ export async function handleGet(path) {
 };
 
 export async function handlePost(path, data) {
+    console.log(path)
```
Comment (EA171) [inline client-side/src/axios/middleware.js:12]: please remove all console log from your code

File: client-side/src/services/profileService.js [hunk 30]
```diff
@@ -9,15 +9,18 @@ export const getAllProfiles = async () => {
         throw err;
     }
 };
+
 export const createProfile = async (profileData) => {
     try {
+        console.log(profileData)
```
Comment (EA171) [inline client-side/src/services/profileService.js:15]: please remove all console log from your code

**Review by EA171 — CHANGES_REQUESTED** [review]

File: client-side/src/components/profileComponents/profilePageComponent.jsx [hunk 31]
```diff
@@ -0,0 +1,231 @@
...
+import { useAppSelector } from '../../redux/store.jsx';
+import { setProfiles, updateProfile } from '../../redux/profile/profile.slice.js';
+import { selectProfile } from '../../redux/profile/profile.selector.js';
+import DeleteIcon from '@mui/icons-material/Delete';
+import EditIcon from '@mui/icons-material/Edit';
+import SaveIcon from '@mui/icons-material/Save';
+import CancelIcon from '@mui/icons-material/Cancel';
+import AddProfileComponent from './addProfileComponent.jsx';
+import UpdateProfileComponent from './updateProfileCpmponent.jsx';
+import ProfileActivationTimer from './profileActivationComponent.jsx';
```
Comment (e171271) [inline client-side/src/components/profileComponents/profilePageComponent.jsx:16]: please remove unused imports

File: client-side/src/components/profileComponents/updateProfileCpmponent.jsx [hunk 32]
```diff
@@ -0,0 +1,101 @@
...
+
+    const handleClickOpen = useCallback(() => {
+        setOpen(true);
+    }, []);
+
+    const handleClose = useCallback(() => {
+        setOpen(false);
+    }, []);
+
+    const handleSave = useCallback(async () => {
```
Comment (e171271) [inline client-side/src/components/profileComponents/updateProfileCpmponent.jsx:46]: why did you add useCallback for each function?

**Review by e171271 — CHANGES_REQUESTED** [review]

File: client-side/src/stories/table/TableComponent.jsx [hunk 33]
```diff
@@ -1,38 +1,326 @@
...
+// };
+// export default TableComponent;
+
 import React from 'react';
 import { DataGrid } from '@mui/x-data-grid';
-import PropTypes from 'prop-types'
+import PropTypes from 'prop-types';
+import IconButton from '@mui/material/IconButton';
+import TextField from '@mui/material/TextField';
+import Select from '@mui/material/Select';
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:167]: Connect all the mui imports into one

File: client-side/src/stories/table/TableComponent.jsx [hunk 34]
```diff
@@ -1,38 +1,326 @@
...
+    value: PropTypes.string.isRequired,
+    text: PropTypes.string.isRequired,
+  })).isRequired,
+  addButton: PropTypes.bool,
+  handleAddRow: PropTypes.func,
+  pageSize: PropTypes.number, 
 };
 
-export default TableComponent;
+TableComponent.defaultProps = {
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:316]: remove defaultPrors and add defaultValues instead

File: client-side/src/stories/table/TableComponent.jsx [hunk 35]
```diff
@@ -1,38 +1,326 @@
...
+  let columns = dataObject.headers.map((header, i) => ({
     field: header,
     headerName: header,
     width: widthOfColums[i],
-    align:'center',
-    headerAlign: 'center'
+    align: 'center',
+    headerAlign: 'center',
+    renderCell: (params) => {
+      if (editRowId && editRowId === params.row.id && header !== 'Actions') {
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:192]: move the renderCell into seperate function

File: client-side/src/stories/table/TableComponent.jsx [hunk 36]
```diff
@@ -1,38 +1,326 @@
...
+        if (header === 'status') {
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:193]: the table is a generic comoponent the types couldn't be according to the name of a column, each column should have a type and this will be the identifier to know if the column should be select/textField etc.

also, please change else, if to switch case

File: client-side/src/stories/table/TableComponent.jsx [hunk 37]
```diff
@@ -1,38 +1,326 @@
...
+                <MenuItem key={option.value} value={option.value}>
+                  {option.text}
+                </MenuItem>
+              ))}
+            </Select>
+          );
+        }
+        if (header === 'name') {
+          return (
+            <Tooltip title='This field is optional'>
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:211]: move the title text to constants file

File: client-side/src/stories/table/TableComponent.jsx [hunk 38]
```diff
@@ -1,38 +1,326 @@
...
+      }
+      return params.value;
+    },
   }));
 
+  if (!columns.some(column => column.field === 'Actions')) {
+    columns.push({
+      field: 'Actions',
+      headerName: 'Actions',
+      width: widthOfColums[widthOfColums.length - 1],
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:251]: if the actions is not exists in the columns, how do you have the width in the widthOfColumns array?

File: client-side/src/stories/table/TableComponent.jsx [hunk 39]
```diff
@@ -1,38 +1,326 @@
...
+            </Tooltip>
+          ))}
+        </div>
+      ),
+    });
+  }
+
   return (
-    <div className="table" style={{ width: widthOfTable ,marginTop:"8%"}}>
+    <div className="table" style={{ width: widthOfTable, marginTop: "8%" }}>
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:269]: move styles to scss file - do it in all this file

File: client-side/src/stories/table/TableComponent.jsx [hunk 40]
```diff
@@ -1,38 +1,326 @@
...
           },
         }}
-        pageSizeOptions={[4, 8]}
+        pageSizeOptions={[4, 8, 10]} 
+        pageSize={pageSize || 5} 
       />
+      {addButton && (
+        <div style={{ textAlign: 'center', marginTop: '10px' }}>
+          <GenericButton variant="outlined" size="medium" onClick={handleAddRow} className="profile-list-button" label='Add Website' />
+        </div>
```
Comment (e171271) [inline client-side/src/stories/table/TableComponent.jsx:284]: labels should be in constants file

File: client-side/src/styles/profileActivationTimer.scss [hunk 41]
```diff
@@ -0,0 +1,57 @@
...
+.timer-label {
+  font-size: 50%;
+  margin-top: 4%;
+  color: $secondary-color;
+}
+
+.countdown-circle-timer {
+  stroke-width: 0.25%;
+  size: 6.25%;
+  //trail-color: $standardInfoColor; // Use standard info color for trail
```
Comment (e171271) [inline client-side/src/styles/profileActivationTimer.scss:36]: remove comments

File: client-side/src/styles/profileActivationTimer.scss [hunk 42]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+.profile-list-container {
+  .profile-timer-button {
+    background-color: transparent;
+    border: 2px solid $secondary-color;
+    color: $secondary-color;
+    padding: 10px 20px;
```
Comment (e171271) [inline client-side/src/styles/profileActivationTimer.scss:44]: don't use px's

File: client-side/src/styles/profilePageStyle.scss [hunk 43]
```diff
@@ -0,0 +1,127 @@
...
+  margin-bottom: 2vh;
+  margin: 2vh;
+  right: 0.7vw;
+}
+.profile-list-container {
+  .profile-list-button {
+    background-color: transparent;
+    border: 2px solid $secondary-color;
+    color: $secondary-color;
+    padding: 10px 20px;
```
Comment (e171271) [inline client-side/src/styles/profilePageStyle.scss:101]: don't use px's

File: client-side/src/styles/profilePageStyle.scss [hunk 44]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+.profile-select {
+  width: 10vw;
+}
+.timer {
+  margin-left:3vh;
+}
+.green_title{
+  color:  rgb(103, 252, 210) ;
```
Comment (e171271) [inline client-side/src/styles/profilePageStyle.scss:126]: colors should always code from variables file

File: client-side/src/styles/profilePageStyle.scss
Comment (e171271) [inline client-side/src/styles/profilePageStyle.scss:1]: this file looks like e css file, please use scss hierarchy

File: client-side/src/utils/profileUtil.js [hunk 45]
```diff
@@ -0,0 +1,123 @@
...
+
+
+export const updateFormDataWithStatusBlockedSites = (formData, value) => {
+    const updatedWebsites = formData.websites.map(website => {
+        if (website.status === 'limit') {
+            return website;
+        }
+        return {
+            ...website,
+            status: value === 'black list' ? 'open' : 'block'
```
Comment (e171271) [inline client-side/src/utils/profileUtil.js:32]: you should have an enum for those statuses

File: client-side/test/components/profileComponents/ProfileActions.test.js [hunk 46]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import ProfileActions from '../../../src/components/profileComponents/profileActions.jsx';
+import '@testing-library/jest-dom';
```
Comment (e171271) [inline client-side/test/components/profileComponents/ProfileActions.test.js:4]: re-order the imports

File: client-side/test/components/profileComponents/ProfileActivationTimer.test.js [hunk 47]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import ProfileActivationTimer from '../../../src/components/profileComponents/profileActivationTimer.jsx';
+import '@testing-library/jest-dom';
```
Comment (e171271) [inline client-side/test/components/profileComponents/ProfileActivationTimer.test.js:4]: re-order the imports

File: client-side/test/components/profileComponents/ProfileForm.test.js [hunk 48]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import ProfileForm from '../../../src/components/profileComponents/profileForm.jsx';
+
+describe('ProfileForm', () => {
+    test('renders profile name input', () => {
```
Comment (e171271) [inline client-side/test/components/profileComponents/ProfileForm.test.js:7]: you can move the initial formData value and the render into beforeEach function

File: client-side/test/components/profileComponents/TimerActivationButton.test.js [hunk 49]
```diff
@@ -0,0 +1,56 @@
...
+
+test('does not start timer with empty input', () => {
+  render(<TimerActivationButton profileName="Profile 1" />);
+
+  fireEvent.click(screen.getByText(BUTTON_LABELS.TIMER_ACTIVATION));
+
+  fireEvent.click(screen.getByText(BUTTON_LABELS.START));
+
+  expect(screen.queryByText('1:00:00')).not.toBeInTheDocument();
+});
```
Comment (e171271) [inline client-side/test/components/profileComponents/TimerActivationButton.test.js:30]: you should chekck according to error text that should be shown.
'something like end time is missing'

File: client-side/test/components/profileComponents/TimerActivationButton.test.js
Comment (e171271) [inline client-side/test/components/profileComponents/TimerActivationButton.test.js:1]: format the file

File: client-side/test/components/profileComponents/UpdateProfileComponent.test.js [hunk 50]
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import { SnackbarProvider } from 'notistack';
+import { BrowserRouter as Router } from 'react-router-dom';
+import UpdateProfileComponent from '../../../src/components/profileComponents/updateProfileCpmponent.jsx';
+import '@testing-library/jest-dom';
```
Comment (e171271) [inline client-side/test/components/profileComponents/UpdateProfileComponent.test.js:8]: re-order the imports

File: client-side/test/utils/profileUtil.test.js [hunk 51]
```diff
@@ -0,0 +1,234 @@
...
+        const formData = {
+            timeProfile: {
+                timeStart: '08:00',
+                timeEnd: '17:00'
+            }
+        };
+        expect(validateProfileDate(formData)).toBe(true);
+    });
+
+    test('returns true if start time is the same as end time', () => {
```
Comment (e171271) [inline client-side/test/utils/profileUtil.test.js:225]: add test for case that timeStrt or timeEnd are missing

File: node_modules/husky/package.json
Comment (e171271) [inline node_modules/husky/package.json:1]: please add node_modules folder to gitIgnore.

File: server-side/app.js [hunk 52]
```diff
@@ -2,19 +2,27 @@ import express from 'express';
 import morgan from 'morgan'
 import cors from 'cors'
 import dotenv from 'dotenv'
+<<<<<<< HEAD
```
Comment (e171271) [inline server-side/app.js:5]: solve the conflicts

File: server-side/services/profie.service.js [hunk 53]
```diff
@@ -0,0 +1,59 @@
...
+        return profiles;
+    } catch (err) {
+        throw new Error(err.message);
+    }
+};
+
+export const createProfile = async (profileData) => {
+    const newProfile = new Profiles(profileData);
+    try {
+        if(newProfile.statusBlockedSites == 'black list'){
```
Comment (e171271) [inline server-side/services/profie.service.js:15]: why this empty if block?

**Review by e171271 — CHANGES_REQUESTED** [review]
I moved over most of the code, but part of them have conflicts, so please solve the conflicts and then I'll be able to finish the review.
